x-pack/plugins/infra/public/containers/logs/log_position/with_log_position_url_state.tsx

```diff
@@ -9,60 +9,106 @@ import React, { useContext, useMemo } from 'react';
 import { pickTimeKey } from '../../../../common/time';
 import { replaceStateKeyInQueryString, UrlStateContainer } from '../../../utils/url_state';
 import { LogPositionState, LogPositionStateParams } from './log_position_state';
+import { isValidDatemath, datemathToEpochMillis } from '../../../utils/datemath';
 
 /**
  * Url State
  */
-
 interface LogPositionUrlState {
   position: LogPositionStateParams['visibleMidpoint'] | undefined;
   streamLive: boolean;
+  start?: string;
+  end?: string;
 }
 
+const ONE_HOUR = 86400000;
+
 export const WithLogPositionUrlState = () => {
   const {
     visibleMidpoint,
-    isAutoReloading,
+    isStreaming,
     jumpToTargetPosition,
-    jumpToTargetPositionTime,
     startLiveStreaming,
     stopLiveStreaming,
+    startDateExpression,
+    endDateExpression,
+    updateDateRange,
+    initialize,
   } = useContext(LogPositionState.Context);
   const urlState = useMemo(
     () => ({
       position: visibleMidpoint ? pickTimeKey(visibleMidpoint) : null,
-      streamLive: isAutoReloading,
+      streamLive: isStreaming,
+      start: startDateExpression,
+      end: endDateExpression,
     }),
-    [visibleMidpoint, isAutoReloading]
+    [visibleMidpoint, isStreaming, startDateExpression, endDateExpression]
   );
   return (
     <UrlStateContainer
       urlState={urlState}
       urlStateKey="logPosition"
       mapToUrlState={mapToUrlState}
       onChange={(newUrlState: LogPositionUrlState | undefined) => {
-        if (newUrlState && newUrlState.position) {
+        if (!newUrlState) {
+          return;
+        }
+
+        if (newUrlState.start || newUrlState.end) {
+          updateDateRange({
+            startDateExpression: newUrlState.start,
+            endDateExpression: newUrlState.end,
+          });
+        }
+
+        if (newUrlState.position) {
           jumpToTargetPosition(newUrlState.position);
         }
-        if (newUrlState && newUrlState.streamLive) {
+
+        if (newUrlState.streamLive) {
           startLiveStreaming();
-        } else if (
-          newUrlState &&
-          typeof newUrlState.streamLive !== 'undefined' &&
-          !newUrlState.streamLive
-        ) {
+        } else if (typeof newUrlState.streamLive !== 'undefined' && !newUrlState.streamLive) {
           stopLiveStreaming();
         }
       }}
       onInitialize={(initialUrlState: LogPositionUrlState | undefined) => {
-        if (initialUrlState && initialUrlState.position) {
-          jumpToTargetPosition(initialUrlState.position);
-        } else {
-          jumpToTargetPositionTime(Date.now());
-        }
-        if (initialUrlState && initialUrlState.streamLive) {
-          startLiveStreaming();
+        if (initialUrlState) {
+          const initialPosition = initialUrlState.position;
+          let initialStartDateExpression = initialUrlState.start || 'now-1d';
```

Comment:
Should the default range depend on the `initialPosition` if that is set? Otherwise the target might not be reachable of older than a day. It could be half a day before and after the position, for example.

On a related note, do we want to add the ability to specify the `from` and `to` in the `link-to` log routes the same way it is done for the `node-detail` metric routes? We can do that in a follow-up PR, of course.

Reply:
> Should the default range depend on the initialPosition if that is set? [...] It could be half a day before and after the position, for example.

Mmm, good point. [What it's doing right now](https://github.com/afgomez/kibana/blob/1e42ffa222c475f94c199f935bf2303167259ffb/x-pack/plugins/infra/public/containers/logs/log_position/with_log_position_url_state.tsx#L88-L94
) is to change the start/end edges if the `initialPosition` doesn't fall within the range (in the default case, before one day).

But it might make sense indeed to _not_ set a default if there's a initial position present _and_ there is no start or end edges.

----

> On a related note, do we want to add the ability to specify the from and to in the link-to log routes the same way it is done for the node-detail metric routes?

Makes sense :+1: